### ISSUE TITLE
"Hide before" hides from menus, even after the specified time

### ISSUE THREAD

**Issue by Discostu36** [neos/neos-development-collection#3823]
Hello,

<strike>after our update to Neos CMS 8.0.2</strike>, The feature of hiding a page until a specific DateTime doesn't work correctly <strike>any more</strike>.  When the moment has passed, the page is published and can be visited. But it is not shown in menus. Only after the entry in the "Hide before" field has been deleted, the page appears in menus.

System log does not show any errors.

# How to reproduce
1. Create a page and a menu where this page would be visible. 
2. Set a "Hide before" time in metadata → visibility for the page
3. Publish the page
4. Wait until the specified time has arrived
5. See that the page is now visible but not in the menu
6. Delete the "hide before" time and publish again
7. See that the page has now appeared in the menu

**Comment by Discostu36** [comment]
I just tried with a fresh installation of 8.0.3 and could reproduce the issue. So this really seems to be a general problem, not something related to our current installation.

**Comment by kitsunet** [comment]
Yes, it sounds very much as if the menu does not include the hide date in the cache lifetime, it's surprising that this is a new issue though, because due to the complexity I was under the impression that we never did for the default menu configuration and that you would have to do that in your own configuration since ages.

**Comment by Discostu36** [comment]
It depends on your definition of a "new issue", but I cannot reproduce the problem with Neos 5.0.4. Don't know about 7.3, maybe I can try later.

**Comment by rolandschuetz** [comment]
The following would fix the general case:
```
@context.startingPoint = ${startingPoint}
@cache {
    mode = 'cached'
    entryIdentifier {
        documentNode = ${documentNode}
    }
    entryTags {
        1 = ${Neos.Caching.nodeTypeTag('Neos.Neos:Document', documentNode)}
    }
    maximumLifetime = ${q(startingPoint || site).context({'invisibleContentShown': true}).find('[instanceof Neos.Neos:Document]').cacheLifetime()}
}
```

A clean solution needs to also consider `filter`maybe not just including document nodes

**Comment by Discostu36** [comment]
@rolandschuetz Thanks, where does this have to be added, on the pages that should appear in the menu?

**Comment by Benjamin-K** [comment]
@Discostu36 This belongs to the Menu.
